dvc/repo/experiments/rename.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from typing import TYPE_CHECKING, Dict, List, Optional, Union
+
+from dvc.scm import Git
+
+from .refs import ExpRefInfo
+from .utils import exp_refs_by_rev
+
+if TYPE_CHECKING:
+    from dvc.repo import Repo
+    from dvc.repo.experiments.queue.celery import LocalCeleryQueue
+
+    from .queue.base import ExpRefAndQueueEntry
+
+
+logger = logging.getLogger(__name__)
+
+
+def rename(
+    repo: "Repo",
+    new_name: str,
+    exp_name: Union[str, None] = None,
+    rev: Optional[Union[List[str], str, None]] = None,
+    git_remote: Optional[str] = None,
+) -> List[str]:
+    assert isinstance(repo.scm, Git)
+    renamed: List[str] = []
+    celery_queue: LocalCeleryQueue = repo.experiments.celery_queue
+
+    if exp_name:
+        results: Dict[
+            str, ExpRefAndQueueEntry
+        ] = celery_queue.get_ref_and_entry_by_names(exp_name, git_remote)
```

Comment:
I think we could just use `from dvc.repo.experiments.utils import resolve_name` instead of this method

Reply:
You're right. I've updated this